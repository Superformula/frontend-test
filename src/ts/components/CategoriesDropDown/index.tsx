// Imports
// -------

// Libraries
import React, {useCallback, useState} from 'react';
// Components
import DropDownMenu from '../DropDownMenu';
// GraphQL
import getCategoriesQuery from "../../graphql/yelp/queries/getCategories";
// Hooks
import requestData from "../../hooks/requestData";
// Utils
import formatCategories from "../../utils/formatCategories";

// Internal
// --------

export interface CategoriesDropDownProps {
    children: Array<string>
}

const CategoriesDropDown: React.FC<CategoriesDropDownProps> = ({ children, setCategories, categorySelected, categorySelectedCallback }) => {
    //
    // State

    const [categoriesMenuSelected, setCategoriesMenuSelected] = useState(false);

    //
    // Callback

    const categoryMenuSelectedCallback = useCallback(() => {
        setCategoriesMenuSelected(!categoriesMenuSelected);
    }, [categoriesMenuSelected, setCategoriesMenuSelected]);

    //
    // Render

    const categoriesData = requestData(getCategoriesQuery, { formatData: formatCategories });
    if (!categoriesData.isLoading && !categoriesData.hasError) {
        setTimeout(() => setCategories(categoriesData), 0);
    }

    return (
        <DropDownMenu text="Categories" selected={ categoriesMenuSelected } onClick={ categoryMenuSelectedCallback } dimensions={{ width: 244, height: 32 }} option={ categorySelected } setOption={ categorySelectedCallback }>
            { children }
        </DropDownMenu>
    );
}

// Exports
// -------

export default CategoriesDropDown;
