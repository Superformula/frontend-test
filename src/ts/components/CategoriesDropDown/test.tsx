// Imports
// -------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import CategoriesDropDown from './index';

// Mocks
// -----

jest.mock("../DropDownMenu", () => 'DropDownMenu');
jest.mock("../../hooks/requestData", () => () => 1);

// Internal
// --------

const children = ['Cat1', 'Cat2', 'Cat3'];

const getComponent = (props = {}) => renderer.create(
    <CategoriesDropDown {...props} >
        { children }
    </CategoriesDropDown>,
);

// Tests
// -----

describe('src/ts/components/CategoriesDropDown', () => {
    let component;

    describe('Rendering', () => {
        it('renders default correctly', () => {
            component = getComponent();
            expect(component).toMatchSnapshot();
        });
    });
});
