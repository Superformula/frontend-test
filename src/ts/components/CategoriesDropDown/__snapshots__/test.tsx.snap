// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/ts/components/CategoriesDropDown Rendering renders default correctly 1`] = `
<DropDownMenu
  dimensions={
    Object {
      "height": 32,
      "width": 244,
    }
  }
  onClick={[Function]}
  selected={false}
  text="Categories"
>
  Cat1
  Cat2
  Cat3
</DropDownMenu>
`;
